funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -550,7 +551,11 @@ def _queue_management_worker(self):
                             # We set all current tasks to this exception to make sure that
                             # this is raised in the main context.
                             for task in self.tasks:
-                                self.tasks[task].set_exception(self._executor_exception)
+                                try:
+                                    self.tasks[task].set_exception(self._executor_exception)
+                                except concurrent.futures.InvalidStateError:
+                                    # Task was already cancelled, the exception can be ignored
+                                    logger.debug(f"Task:{tid} result couldn't be set. Already in terminal state")
```

Comment:
I think this {tid} is not a task ID - in this branch of the `if`, `tid == -1`

Reply:
You are right, I've renamed the loop var to `task_id` and used that.